thorlcr/master/thdemonserver.cpp

```diff
@@ -102,39 +102,21 @@ class DeMonServer : public CSimpleInterface, implements IDeMonServer
     unsigned lastReportedMsTick;
     unsigned reportRateMSecs;
     CIArrayOf<CGraphBase> activeGraphs;
-    UnsignedArray graphStartsMsTick;
-    double thorManagerRate = 0;
-    double thorWorkerRate = 0;
     cost_type costLimit = 0;
     cost_type previousExecutionCost = 0;
     StatisticsAggregator statsAggregator;
 
-    void reportStatus(IWorkUnit *wu, CGraphBase &graph, unsigned startTime, bool finished, bool success)
+    void reportStatus(IWorkUnit *wu, CGraphBase &graph, bool finished, bool success)
     {
         const char *graphName = graph.queryJob().queryGraphName();
         unsigned wfid = graph.queryJob().getWfid();
         StringBuffer timer, graphScope;
         formatGraphTimerLabel(timer, graphName, 0, graph.queryGraphId());
         formatGraphTimerScope(graphScope, wfid, graphName, 0, graph.queryGraphId());
-        unsigned duration = msTick()-startTime;
+        // Use the elapsed time tracked in CMasterGraph (using ::getLastElapsedCycles) as the cost calculated for
+        // subgraph was based on that elapsed time
+        unsigned duration = cycle_to_millisec(graph.getLastElapsedCycles());
```

Comment:
This should be converting and reporting in nanoseconds.  (I was about to open a jira to request that change.)